lib/api/src/common/reactors.rs

```diff
@@ -0,0 +1,88 @@
+use std::collections::VecDeque;
+use std::sync::Mutex;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::mpsc;
+use std::sync::Arc;
+use std::task::Waker;
+use std::time::Duration;
```

Comment:
This doesn't need to live inside of Wasmer. It can be a side utility/module inside of wasmer-wasi atm (since it's only used there)

Reply:
I have indeed now moved this to wasmer-wasi instead